nuxeo-services/nuxeo-platform-filemanager-core/src/test/java/org/nuxeo/ecm/platform/filemanager/TestCSVZipImporter.java

```diff
@@ -0,0 +1,197 @@
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.platform.filemanager.api.FileImporterContext;
+import org.nuxeo.ecm.platform.filemanager.api.FileManager;
+import org.nuxeo.ecm.platform.filemanager.service.extension.CSVZipImporter;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertSame;
+
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@RepositoryConfig(init = RepositoryInit.class, cleanup = Granularity.METHOD)
+@Deploy("org.nuxeo.ecm.platform.types.api")
+@Deploy("org.nuxeo.ecm.platform.types.core")
+@Deploy("org.nuxeo.ecm.platform.filemanager.core")
+@Deploy("org.nuxeo.ecm.platform.filemanager.core.tests:faceted-tag-core-types-contrib.xml")
+public class TestCSVZipImporter {
+
+    @Inject
+    protected CoreSession coreSession;
+
+    protected DocumentModel workspace1;
+
+    protected DocumentModel workspace2;
+
+    protected DocumentModel wsRoot;
+
+    private static File getArchiveFile(String file) {
+        return new File(FileUtils.getResourcePathFromContext(file));
+    }
+
+    @Before
+    public void createTestDocuments() throws Exception {
+        wsRoot = coreSession.getDocument(new PathRef("/default-domain/workspaces"));
+
+        workspace1= coreSession.createDocumentModel(wsRoot.getPathAsString(), "ws1", "Workspace");
+        workspace1.setProperty("dublincore", "title", "test WS1");
+        workspace1 = coreSession.createDocument(workspace1);
+
+        workspace2= coreSession.createDocumentModel(wsRoot.getPathAsString(), "ws2", "Workspace");
+        workspace2.setProperty("dublincore", "title", "test WS2");
+        workspace2 = coreSession.createDocument(workspace2);
+    }
+
+    @Test
+    public void testArchiveDetection() throws Exception {
+        ZipFile archive = CSVZipImporter.getArchiveFileIfValid(getArchiveFile("test-data/testCSVArchive.zip"));
+        assertNotNull(archive);
+        archive.close();
+    }
+
+    @Test
+    public void testImportViaFileManager() throws Exception {
+        File archive = getArchiveFile("test-data/testCSVArchive.zip");
+        FileManager fm = Framework.getService(FileManager.class);
+        Blob blob = Blobs.createBlob(archive);
+        FileImporterContext context = FileImporterContext.builder(coreSession, blob, workspace1.getPathAsString())
+                                                         .overwrite(true)
+                                                         .build();
+        fm.createOrUpdateDocument(context);
+        DocumentModelList children = coreSession.getChildren(workspace1.getRef());
+        assertSame(2, children.size());
```

Comment:
you shoud assertEquals on this. assertSame is to check reference equality. So the arguments are boxed without a reason.

Reply:
updated to assertEquals